keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/diff/PartialDerivatives.java

```diff
@@ -418,10 +413,24 @@ public PartialDerivatives slice(int dimension, long index, boolean reshape) {
     }
 
     private long[] extractWrtShape(long[] partialDerivativeShape, int rankOfSource) {
+        if (partialDerivativeShape.length == 0) {
```

Comment:
are these new paths covered in testing?

Reply:
Yes. Many times. Any time there's a diff of a scalar wrt to a scalar this path is taken.

Reply:
👍